### PR TITLE
fix update password and login/signup navigate

### PR DESCRIPTION
**Pull request by Kirazuto7**
Changes:

Login & Signup: Fix the navigation to the homepage after sign in. Also set the props for detecting navigation to true so the page loads with the current user information from App.js

Profile.js: Only show update password for email&password users otherwise hide it for google sign in users

UpdatePassword.jsx: Make the user reauthenticate with their current password before changing their password to a new password and fixed any errors with updating the password

### CODE REVIEW

File: my-app/src/Components/App.js
Comment (chizy7) [inline my-app/src/Components/App.js:1]: Looks good.

File: my-app/src/Components/Profile.js
Comment (chizy7) [inline my-app/src/Components/Profile.js:1]: The updates look good.

File: my-app/src/Components/ProfilePage/UpdatePassword.jsx
Comment (chizy7) [inline my-app/src/Components/ProfilePage/UpdatePassword.jsx:1]: This looks good too and addresses the issues you mentioned.

File: my-app/src/Components/Signup.js
Comment (chizy7) [inline my-app/src/Components/Signup.js:1]: Looks good too.

**Review by chizy7 — APPROVED** [review]
This looks good and addresses the issues pointed out. This can be merged after @caid4134 reviews.

**Review by caid4134 — APPROVED** [review]
Fixes look good for maintenance.

**Pull request merged**